objects/ui2/ui2Event.self

```diff
@@ -550,7 +550,7 @@ Feel free to inherit me and override the ones you can implement.
             | 
             combo nonmodifierKeyCap isArrow ifTrue: [^ handleSmallArrowPress: combo IfCannot: b].
 
-            combo nonmodifierKeyCap isPrintable ifTrue: [^combo nonmodifierKeyCap printString do: [|:c| insert_char: c]].
+            combo nonmodifierKeyCap printString size <= 6 ifTrue: [^combo nonmodifierKeyCap printString do: [|:c| insert_char: c]].
```

Comment:
If I am looking at it right, to keep the original behaviour, you want to decode the first codepoint and apply `>= 0x20 && < 0x80 || >= 0xa0` to that to reject C0 and C1 control characters at least.

It's also just going to break in certain cases as not everything fits in one codepoint, even though it probably shouldn't happen with normal characters that people use.

The original code makes me wonder what it receives on Ctrl-letter and Alt-letter though, whether it does the same as your typical terminal and forces the highest bits to 000, or 1 respectively.